worker/upgradeseries/upgrader.go

```diff
@@ -114,17 +114,9 @@ func (u *upgrader) ensureSystemdFiles() error {
 		return nil
 	}
 
-	services, links, failed, err := u.manager.WriteSystemdAgents(
-		u.machineAgent, u.unitAgents, paths.NixDataDir, systemdMultiUserDir)
-
-	if len(services) > 0 {
-		u.logger.Infof("agents written and linked by systemd: %s", strings.Join(services, ", "))
-	}
-	if len(links) > 0 {
-		u.logger.Infof("agents written and linked by symlink: %s", strings.Join(links, ", "))
-	}
-
-	return errors.Annotatef(err, "failed to write agents: %s", strings.Join(failed, ", "))
+	return errors.Annotatef(
+		u.manager.WriteSystemdAgents(u.machineAgent, paths.NixDataDir, systemdMultiUserDir),
+		"failed to write machine agent")
```

Comment:
I always annotate these as "writing machine agent", the "failed" part being implied by the non-nil error.

Reply:
Sounds like a good idea.